### PR TITLE
Fix a few compilation problems that occur when using c++14 mode

### PR DESCRIPTION
**Pull request by biddisco**
make_unique has been moved into nest:mc::util namespace to avoid a conflict with std::make_unique
util:: cbegin and cend clash with std:: equivalents, so we call them by namespace explicitly.

### CODE REVIEW

File: miniapp/miniapp.cpp [hunk 1]
```diff
@@ -224,7 +224,7 @@ std::unique_ptr<sample_trace_type> make_trace(cell_member_type probe_id, probe_s
     }
     name += probe.location.segment? "dend" : "soma";
 
-    return util::make_unique<sample_trace_type>(probe_id, name, units);
+    return nest::mc::util::make_unique<sample_trace_type>(probe_id, name, units);
```
Comment (bcumming) [inline miniapp/miniapp.cpp]: It isn't necessary to use the full namespace `nest::mc::` is it? This is very verbose.

**Comment by bcumming** [comment]
I think that these changes are good, because they disambiguate some things, and will be useful when we finally move to C++14. I would like to see the short form, i.e. `util::xxx` instead of `nest::mc::util::xxx` used wherever possible however.

**Comment by biddisco** [comment]
I did a search/replace and I'll put things back to just util:: namespace and push a new version.

The reason or c++14 is simply that in my CMake Superproject, I'm using some c++ 14 features in hpx and to compile everything in one go, I am using the same flags everywhere. I would not put a dependency on c++14 into nestmc.

**Comment by biddisco** [comment]
I've force pushed a new version without the nest::mc::util

**Comment by bcumming** [comment]
looks good to me.

**Pull request merged**